internal/runtime/handler.go

```diff
@@ -205,13 +219,31 @@ func (h *Handler) getRuntimes(w http.ResponseWriter, req *http.Request) {
 			httputil.WriteErrorResponse(w, http.StatusInternalServerError, err)
 			return
 		}
-		err = h.setRuntimeOptionalAttributes(&dto, kymaConfig, clusterConfig, gardenerConfig)
+
+		instanceDrivenByKimOnly := h.kimConfig.IsDrivenByKimOnly(dto.ServicePlanName)
+
+		err = h.setRuntimeOptionalAttributes(&dto, kymaConfig, clusterConfig, gardenerConfig, instanceDrivenByKimOnly)
 		if err != nil {
 			h.logger.Warn(fmt.Sprintf("unable to set optional attributes: %s", err.Error()))
 			httputil.WriteErrorResponse(w, http.StatusInternalServerError, err)
 			return
 		}
 
+		if runtimeResourceConfig && dto.RuntimeID != "" && instanceDrivenByKimOnly {
+			runtimeResourceName, runtimeNamespaceName := h.getRuntimeNamesFromLastOperation(dto)
+
+			runtimeResourceObject := &unstructured.Unstructured{}
+			runtimeResourceObject.SetGroupVersionKind(RuntimeResourceGVK())
+			err = h.k8sClient.Get(context.Background(), client.ObjectKey{
+				Namespace: runtimeNamespaceName,
+				Name:      runtimeResourceName,
+			}, runtimeResourceObject)
+			if err != nil {
```

Comment:
I think in case of "Not Found" error we will have a problem here.

Reply:
If there is no runtime resource, let's do not throw any error, just leave the field empty.

Reply:
We log Warning and return empty result.

Reply:
handle NotFoundError -> log.Info, other error log as warn

Reply:
In the next commit/push.